src/kixi/hecuba/minorca.clj

```diff
@@ -112,29 +109,34 @@
   [& args]
   (let [{:keys [project-id embed-url username password] :as opts}
         (:options (parse-opts args cli-options))
-        {:keys [mapping-file processed-file]} config-info
-        s3-files (list-files-in-bucket)]
-    (log/info "S3 files:" s3-files)
-    (doseq [f s3-files]
-      (println "> FILE " f)
-      (let [file-info (s3/get-object cred bucket f)
-            input-data (with-open [r (->> (s3/get-object cred bucket f)
-                                          :object-content
-                                          io/reader)]
-                         (pro/file->seq-of-maps r))
-            processed-content (with-open [in-file (io/reader processed-file)]
-                                (vec (csv/read-csv in-file)))
-            data-to-save (conj (vec processed-content)
-                               [(f/unparse fmt (t/now)) (:key file-info) (:bucket-name file-info)
-                                (:object-metadata file-info) (:object-content file-info)])]
-        (println ">> Pre-processing... " f)
-        (pro/write-to-file processed-file data-to-save)
-        (pre-processing input-data
-                        mapping-file project-id
-                        embed-url username password)
-        (println ">> Processing... " f)
-        (upload-measurement-data input-data mapping-file
-                                 embed-url username password)))))
+        {:keys [s3 mapping-file processed-file]} config-info
```

Comment:
I'd add

`{:keys [cred bucket} s3` 

to the let so you can use them in the places in this function